File: triangle.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        int row = triangle.size();
        int[] memo = new int[row];

        for (int i = 0; i < row; i++) {
            memo[i] = triangle.get(row - 1).get(i);
        }

        for (int r = row-2; r >= 0; r--) {
            for (int c = 0; c <= r; c++) {
                memo[c] = Math.min(memo[c], memo[c+1]) + triangle.get(r).get(c);
            }
        }

        return memo[0];        
    }
}

File: triangle.js
/**
 * @param {number[][]} triangle
 * @return {number}
 */
var minimumTotal = function(triangle) {
    let row = triangle.length;
    let memo = [...triangle[row-1]];

    for (let r = row-2; r >= 0; r--) {
        for (let c = 0; c <= r; c++) {
            memo[c] = Math.min(memo[c], memo[c+1]) + triangle[r][c];
        }
    }
    
    return memo[0];
};
